math/binary_exponentiation.c

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file
+ * @brief Calculates the nth power of a number in O(log(n)) time.
+ * @details
+ * Calculating the nth power of a number requires O(n) time using the naive
+ * method. This algorithm optimizes it through exponentiation by squaring.
+ * https://cp-algorithms.com/algebra/binary-exp.html
+ * @author [Praneeth Jain](https://github.com/PraneethJain)
+ */
+
+#include <assert.h>  // for assert()
+#include <stdio.h>   // for output
+#include <stdlib.h>  // for exit()
+
+/**
+ * @brief Determines a raised to the nth power via binary exponentiation.
+ * @param a - base
+ * @param n - exponent
+ * @return a raised to the nth power
+ * @warning
+ * `int` can overflow very quickly.
```

Comment:
use a larger and unsigned type like uint64

Reply:
The base can be negative so switched to int64_t for that, and switched to uint64_t for the exponent.